sdk/file_sdk/api/istream_handler.h

```diff
@@ -0,0 +1,56 @@
+#ifndef ISTREAMHANDLER_H
+#define ISTREAMHANDLER_H
+
+#include <memory>
+
+#include <IStream.h>
+#include <UserPolicy.h>
+#include "label.h"
+#include "fileapi_export.h"
+#include "ipolicy_engine.h"
+#include "labeling_options.h"
+
+using rmscrypto::api::IStream;
+using rmscore::modernapi::UserPolicy;
+using mip::file::ILabel;
+using mip::file::LabelingOptions;
+using mip::file::IPolicyEngine;
+
+namespace mip {
+namespace file {
+
+class IStreamHandler {
+public:
+  static std::shared_ptr<class ILabelAndProtectStream> Create(std::shared_ptr<IPolicyEngine> engine, std::shared_ptr<IStream> inputStream, const std::string& inputExtension);
+
+  virtual bool IsLabeled() = 0;
+
+  virtual bool IsProtected() = 0;
+
+  virtual std::shared_ptr<Tag> GetLabel() = 0;
+
+  virtual void SetLabel(const std::string& labelId, const LabelingOptions& labelingOptions) = 0;
+
+  virtual void SetLabel(std::shared_ptr<ILabel> label, const LabelingOptions& labelingOptions) = 0;
```

Comment:
you can comment this out until we move to the mip repo, if you don't want to add the label.h class

Reply:
already added a stub ILabel

---
In reply to: [138427476](https://github.com/AzureAD/rms-sdk-for-cpp/pull/149#discussion_r138427476) [](ancestors = 138427476)